lib/app_profiler/middleware.rb

```diff
@@ -24,13 +25,31 @@ def call(env, params = AppProfiler::RequestParameters.new(Rack::Request.new(env)
 
     def profile(env, params)
       response = nil
+      orig_backend = nil
 
       return yield unless params.valid?
 
       params_hash = params.to_h
 
       return yield unless before_profile(env, params_hash)
 
+      @backend_lock.synchronize do
+        if params.backend &&
+            AppProfiler.profiler_backend.name.split("::").last.downcase.gsub("backend", "") != params.backend
```

Comment:
I think this validation should move here
https://github.com/Shopify/app_profiler/blob/9eb54130da5adf6f731f2c5aee86fcfebc1e7915/lib/app_profiler/server.rb#L127

Reply:
Maybe in request_params instead? The server code is only for the standalone server

Reply:
https://github.com/Shopify/app_profiler/blob/9eb54130da5adf6f731f2c5aee86fcfebc1e7915/lib/app_profiler/request_parameters.rb#L19 yeah here, sorry I linked the wrong thing.